eng/common/scripts/Invoke-NpmInstall.ps1

```diff
@@ -0,0 +1,66 @@
+. $PSScriptRoot/Helpers/PSModule-Helpers.ps1
+. $PSScriptRoot/Helpers/CommandInvocation-Helpers.ps1
+. $PSScriptRoot/common.ps1
+
+Push-Location $RepoRoot
+try {
+    $currentDur = Resolve-Path "."
+
+    if (Test-Path "node_modules") {
```

Comment:
This condition is not correct:
1. Two processes will install concurrently especially in the pipeline because we run all the projects concurrently.
2. You need to check whether the version installed is the same as what you are going to install, not just check whether npm_modules exists. Otherwise, our customer will never install the next version!

Reply:
I removed the short circuit in this script in favor of an explicit $SkipNpmInstall parameter in thecalling scripts, TypeSpec-Project-Generate

Reply:
The short circuit has been updated with a content test on node_module's internal `.package-lock.json` file.  We also avoid running the install script entirely in concurrent processes by passing $SkipNpmInstall to the calling script, Typespec-Project-Generate.ps1

Reply:
I removed the node_modules check until we decide what checks tsp-client will use